WordPress/Classes/Services/ReaderPostService.m

```diff
@@ -228,68 +232,74 @@ - (void)toggleLikedForPost:(ReaderPost *)post success:(void (^)(void))success fa
             return;
         }
 
-        // Keep previous values in case of failure
-        BOOL oldValue = readerPost.isLiked;
-        BOOL like = !oldValue;
-        NSNumber *oldCount = [readerPost.likeCount copy];
+        [self toggleLikedForPost:readerPost inContext:context success:success failure:failure];
+        [[ContextManager sharedInstance] saveContext:context];
+    }];
+}
 
-        // Optimistically update
-        readerPost.isLiked = like;
-        if (like) {
-            readerPost.likeCount = @([readerPost.likeCount integerValue] + 1);
-        } else {
-            readerPost.likeCount = @([readerPost.likeCount integerValue] - 1);
-        }
-        [[ContextManager sharedInstance] saveContext:self.managedObjectContext];
+- (void)toggleLikedForPost:(ReaderPost *)readerPost inContext:(NSManagedObjectContext *)context success:(void (^)(void))success failure:(void (^)(NSError *error))failure
+{
+    NSParameterAssert(readerPost.managedObjectContext == context);
 
-        NSDictionary *railcar = [readerPost railcarDictionary];
-        // Define success block.
-        NSNumber *postID = readerPost.postID;
-        NSNumber *siteID = readerPost.siteID;
-        void (^successBlock)(void) = ^void() {
-            if (postID && siteID) {
-                NSDictionary *properties = @{
-                                              WPAppAnalyticsKeyPostID: postID,
-                                              WPAppAnalyticsKeyBlogID: siteID
-                                              };
-                if (like) {
-                    [WPAnalytics trackReaderStat:WPAnalyticsStatReaderArticleLiked properties:properties];
-                    if (railcar) {
-                        [WPAnalytics trackReaderStat:WPAnalyticsStatReaderArticleLiked properties:railcar];
-                    }
-                } else {
-                    [WPAnalytics trackReaderStat:WPAnalyticsStatReaderArticleUnliked properties:properties];
+    // Keep previous values in case of failure
+    BOOL oldValue = readerPost.isLiked;
+    BOOL like = !oldValue;
+    NSNumber *oldCount = [readerPost.likeCount copy];
+
+    // Optimistically update
+    readerPost.isLiked = like;
+    if (like) {
+        readerPost.likeCount = @([readerPost.likeCount integerValue] + 1);
+    } else {
+        readerPost.likeCount = @([readerPost.likeCount integerValue] - 1);
+    }
+
+    NSDictionary *railcar = [readerPost railcarDictionary];
```

Comment:
TIL about the _railcar_ concept. 

I don't think it's an industry standard, but rather internal jargon for a kind of object added to Tracks (which I guess is Tracks as in train tracks) to collect additional information.